packages/icons/src/extract.js

```diff
@@ -2,20 +2,64 @@
 const fs = require('fs');
 const path = require('path');
 
-function extractFiles(folder) {
-	const dir = path.join(__dirname, folder);
+function getAbsolutePath(folder) {
+	if (folder.startsWith(__dirname)) {
+		return folder;
+	}
+	return path.join(__dirname, folder);
+}
 
+function getFiles(folder) {
+	const dir = getAbsolutePath(folder);
 	// case insensitive sort for windows users
-	return fs
-		.readdirSync(dir)
-		.sort(function(a, b) {
-			return a.localeCompare(b, { sensitivity: 'base' });
-		})
-		.reduce(
-			(state, file) =>
-				Object.assign(state, { [file.split('.svg')[0]]: fs.readFileSync(path.resolve(dir, file)) }),
-			{},
-		);
+	return fs.readdirSync(dir).sort((a, b) => a.localeCompare(b, { sensitivity: 'base' }));
+}
+
+function getIconId(file) {
+	return file.split('.svg')[0];
+}
+
+function assertUnique(files, acc) {
+	const alreadyExists = Object.keys(files).filter(file => {
```

Comment:
Instead of going over the whole list, you can use Array.some()
```javascript
const hasDuplicate = Object.keys(files).some(file =>  acc[getIconId(file)]);
if (hasDuplicate) {
    // throw
}
```